src/VisualStudio/VisualBasic/Impl/Snippets/SnippetCompletionProvider.vb

```diff
@@ -52,10 +52,14 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Snippets
             Dim syntaxFacts = document.GetLanguageService(Of ISyntaxFactsService)()
             Dim isPossibleTupleContext = syntaxFacts.IsPossibleTupleContext(syntaxTree, position, cancellationToken)
 
-            context.IsExclusive = True
+            context.IsExclusive = ShouldBeExclusive(context.Options)
             context.AddItems(CreateCompletionItems(snippets, isPossibleTupleContext))
         End Function
 
+        Private Function ShouldBeExclusive(options As OptionSet) As Boolean
+            Return not options.GetOption(CompletionOptions.SnippetsBehavior, LanguageNames.VisualBasic) = SnippetsRule.AlwaysInclude
```

Comment:
Why not check in the positive against `SnippetsRule.IncludeAfterTypingIdentifierQuestionTab`, which is the one case where it should be exclusive (I think, right?). Would make this more readable.